wallets/core/src/namespaces/evm/builders.ts

```diff
@@ -0,0 +1,10 @@
+import type { EvmActions } from './types.js';
+
+import { ActionBuilder } from '../../mod.js';
+import { intoConnectionFinished } from '../common/after.js';
+import { connectAndUpdateStateForMultiNetworks } from '../common/and.js';
+
+export const connect = () =>
+  new ActionBuilder<EvmActions, 'connect'>('connect')
+    .and(connectAndUpdateStateForMultiNetworks)
+    .after(intoConnectionFinished);
```

Comment:
I didn't get the point why some hooks are here and some of them should be imported by user in each evm provider. I believe it produces a lot of duplicate codes.

If this is the standard builder, it's better that we could use the final connect builder directly whenever we want instead of duplicating this code in each provider.


```
const connect = builders
  .connect()
  .action(actions.connect(evmPhantom))
  .before(changeAccountSubscriber)
  .or(changeAccountCleanup)
  .build();
```

Reply:
In general, we only keep codes that are same between all providers in standard library which means we will add more functionality to actions like `connect` over time when we are sure about what we are adding is working on all providers. Until then, I prefer to be restrict on what we are adding to core and if we saw a good amount duplication we can use a general solution for them and move the functionality to core.

Reply:
As we talked offline, We decided to move these to `core` action builder. the code lives in my another PR, so I will do the change in that PR alongside phantom.